### PR TITLE
Register requests

### PR DESCRIPTION
**Pull request by CaesiumFox**
Добавлены методы для отклонения заявки, для вывода списка заявок. Добавлена рассылка e-mail при одобрении и отклонении заявки.

### CODE REVIEW

**Comment by IP-13** [comment]
|Overall Project|48.72% **`-4.93%`**|:x:|
|:-|:-|:-:|
|Files changed|2.07%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/433db34be02a29ad5211e56eb179c4c8af237f02/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/433db34be02a29ad5211e56eb179c4c8af237f02/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|12.77% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/433db34be02a29ad5211e56eb179c4c8af237f02/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|10.27% **`-28.42%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|49.07% **`-4.89%`**|:x:|
|:-|:-|:-:|
|Files changed|2.07%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/633b61f57c515d28a5c75d99cdeba9256a112732/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/633b61f57c515d28a5c75d99cdeba9256a112732/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|12.77% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/633b61f57c515d28a5c75d99cdeba9256a112732/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|10.27% **`-28.42%`**|:x:|

File: src/main/java/org/itmo/eventapp/main/service/AuthenticationService.java [hunk 1]
```diff
@@ -76,12 +87,16 @@ public void approveRegistrationRequestCallback(int requestId) {
...
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "Запрос уже обработан");
+        }
+
         UserLoginInfo loginInfo = new UserLoginInfo();
         User user = new User();
         UserNotificationInfo notificationInfo = new UserNotificationInfo();
 
-        var reader = roleRepository.findByName("Читатель")
-                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Роли 'Читатель' не существует"));
+        var reader = roleRepository.findByName(defaultRole)
```
Comment (rimnvd) [inline src/main/java/org/itmo/eventapp/main/service/AuthenticationService.java:98]: Теперь есть RoleService, в нем есть метод getReaderRole()

**Comment by IP-13** [comment]
|Overall Project|53.79% **`-4.13%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|2.07%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/2b3f4a9957e727c3d84ac5d9d8301b1041383450/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/2b3f4a9957e727c3d84ac5d9d8301b1041383450/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|12.77% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/2b3f4a9957e727c3d84ac5d9d8301b1041383450/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|10.27% **`-28.42%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|55.92% **`-4.13%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|2.07%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/e5c2b68dc53ce914a1ca2741488ee27b2a5c3be6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/e5c2b68dc53ce914a1ca2741488ee27b2a5c3be6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|12.77% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/e5c2b68dc53ce914a1ca2741488ee27b2a5c3be6/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|10.27% **`-28.42%`**|:x:|

File: src/main/java/org/itmo/eventapp/main/controller/AuthController.java [hunk 2]
```diff
@@ -32,7 +37,23 @@ ResponseEntity<Void> register(@Valid @RequestBody RegistrationUserRequest regist
 
     @PostMapping(value = "/approveRegister/{requestId}")
     ResponseEntity<Void> approveRegister(@PathVariable("requestId") Integer requestId) {
+        // TODO: check for administrator
         authenticationService.approveRegistrationRequestCallback(requestId);
         return ResponseEntity.status(HttpStatus.OK).build();
     }
+
+    @PostMapping(value = "/declineRegister/{requestId}")
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/controller/AuthController.java:45]: Зачем тут пост?

Comment (IP-13) [reply src/main/java/org/itmo/eventapp/main/controller/AuthController.java:45]: Кажется лучше Put

File: src/main/java/org/itmo/eventapp/main/controller/AuthController.java [hunk 3]
```diff
@@ -32,7 +37,23 @@ ResponseEntity<Void> register(@Valid @RequestBody RegistrationUserRequest regist
...
+    ResponseEntity<Void> declineRegister(@PathVariable("requestId") Integer requestId) {
+        // TODO: check for administrator
+        authenticationService.declineRegistrationRequestCallback(requestId);
+        return ResponseEntity.status(HttpStatus.OK).build();
+    }
+
+    @GetMapping(value = "/listRegisterRequests")
+    ResponseEntity<List<RegistrationRequest>> listRegisterRequests() {
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/controller/AuthController.java:53]: Нельзя просто взять, и отдать RegistrationRequest, где есть пароль, наружу
Нужно добавить дто. RegistrationRequestDto, в которой не будет поля password и отдавать ее

**Comment by IP-13** [comment]
|Overall Project|52.53% **`-3.83%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|1.97%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/1acf7ab678d92b98455e7d2f7e85939bbe2cc1e9/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/1acf7ab678d92b98455e7d2f7e85939bbe2cc1e9/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|12.77% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/1acf7ab678d92b98455e7d2f7e85939bbe2cc1e9/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|10.03% **`-30.1%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|52.46% **`-3.83%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/789c0a20fd2a1412151aa831d284ddf07cfbf917/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/789c0a20fd2a1412151aa831d284ddf07cfbf917/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|12.77% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/789c0a20fd2a1412151aa831d284ddf07cfbf917/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|8.19% **`-30.72%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|59.61% **`-3.57%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/c0f5b7de83115f9494ff8dc8b6e66c2c2ebe2208/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/c0f5b7de83115f9494ff8dc8b6e66c2c2ebe2208/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|12.77% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/c0f5b7de83115f9494ff8dc8b6e66c2c2ebe2208/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|8.19% **`-30.72%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|63.89% **`-3.46%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/6002acb2048a41b6bdf53eee9979c3a3e2e23fbf/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/6002acb2048a41b6bdf53eee9979c3a3e2e23fbf/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|12.77% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/6002acb2048a41b6bdf53eee9979c3a3e2e23fbf/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|8.19% **`-30.72%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|69.27% **`-3.56%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/6d6ac6e8d386575fab9eadb3def1c7d2dacbf6f1/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/6d6ac6e8d386575fab9eadb3def1c7d2dacbf6f1/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|29.79% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/6d6ac6e8d386575fab9eadb3def1c7d2dacbf6f1/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|22.8% **`-33.2%`**|:x:|
|[RegistrationRequestService.java](https://github.com/itmo-events-app/event-app/blob/6d6ac6e8d386575fab9eadb3def1c7d2dacbf6f1/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRegistrationRequestService.java)|12% **`-38%`**|:x:|

**Comment by IP-13** [comment]
|Overall Project|70.57% **`-3.36%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|0%|:x:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[MailSenderServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/a8525de999eb837e616bd19ef643559602f6fe6f/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fmail%2FMailSenderServiceImpl.java)|78.17% **`-21.83%`**|:x:|
|[AuthController.java](https://github.com/itmo-events-app/event-app/blob/a8525de999eb837e616bd19ef643559602f6fe6f/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FAuthController.java)|29.79% **`-34.04%`**|:x:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/a8525de999eb837e616bd19ef643559602f6fe6f/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|22.62% **`-32.94%`**|:x:|
|[RegistrationRequestService.java](https://github.com/itmo-events-app/event-app/blob/a8525de999eb837e616bd19ef643559602f6fe6f/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FRegistrationRequestService.java)|12% **`-38%`**|:x:|

**Review by Darklol — APPROVED** [review]

**Review by IP-13 — APPROVED** [review]

**Pull request merged**